js/wazirx.js

```diff
@@ -0,0 +1,708 @@
+                'limits': limits,
+                'precision': precision,
+                'type': 'spot',
+                'spot': isSpot,
+                'active': active,
+            });
+        }
+        return result;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // [1, 5, 10, 20, 50, 100, 500, 1000]
+        }
+        const response = await this.publicGetDepth (this.extend (request, params));
+        //
+        //     {
+        //          "timestamp":1559561187,
+        //          "asks":[
+        //                     ["8540.0","1.5"],
+        //                     ["8541.0","0.0042"]
+        //                 ],
+        //          "bids":[
+        //                     ["8530.0","0.8814"],
+        //                     ["8524.0","1.4"]
+        //                 ]
+        //      }
+        //
+        const timestamp = this.safeInteger (response, 'timestamp');
+        return this.parseOrderBook (response, symbol, timestamp);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        const ticker = await this.publicGetTicker24hr (this.extend (request, params));
+        //
+        // {
+        //     "symbol":"wrxinr",
+        //     "baseAsset":"wrx",
+        //     "quoteAsset":"inr",
+        //     "openPrice":"94.77",
+        //     "lowPrice":"92.7",
+        //     "highPrice":"95.17",
+        //     "lastPrice":"94.03",
+        //     "volume":"1118700.0",
+        //     "bidPrice":"94.02",
+        //     "askPrice":"94.03",
+        //     "at":1641382455000
+        // }
+        //
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const tickers = await this.publicGetTickers24hr ();
+        //
+        // [
+        //     {
+        //        "symbol":"btcinr",
+        //        "baseAsset":"btc",
+        //        "quoteAsset":"inr",
+        //        "openPrice":"3698486",
+        //        "lowPrice":"3641155.0",
+        //        "highPrice":"3767999.0",
+        //        "lastPrice":"3713212.0",
+        //        "volume":"254.11582",
+        //        "bidPrice":"3715021.0",
+        //        "askPrice":"3715022.0",
+        //     }
+        //     ...
+        // ]
+        //
+        const result = {};
+        for (let i = 0; i < tickers.length; i++) {
+            const ticker = tickers[i];
+            const marketId = this.safeString (ticker, 'symbol');
+            const market = this.safeMarket (marketId, undefined);
```

Comment:
The above two lines – we should move to parseTicker, and take the symbol from the parsed result.

Reply:
@kroitor  done

Reply:
Let's still remove 286-287 and the second argument from 288.
I mean, the point is to parse the symbol inside parseTicker, not outside of it.

Reply:
@kroitor got it now I was thinking that we needed a valid market value to pass to the parseTicker, but we don'ta

Reply:
@carlosmiei yep, in the ideal case a parse-method should be self-sufficient and the exchange should provide enough info to deduce the symbol from the ticker structure itself, and the `market` argument is just a helper for those cases that are far from ideal.